### DISCUSSION TITLE
Can't compile and flash tasmota on m5stack atom lite

### DISCUSSION THREAD

**Discussion by Zarkstar** [arendst/Tasmota#21710] [Support request] [answered]
I can't compile Tasmota correctly and I don't know if I am missing something.
What I did is:

1. downloading Tasmota Master (the same problem occurred with the development branch as well though)
2. configuring user_config_override.h to be able to use Tasmota Neopool
3. added to the end of platformio.ini the code required to see the m5stack environment: 
`[env:m5stack-atoms3]
platform = espressif32
board = m5stack-atoms3
upload_port = /dev/cu.usbserial-2552F4E512`
4. I've selected the environment and launched the complier

It goes on for a bit until I get this message:

In file included from /Users/edo/Downloads/Tasmota-development/tasmota/tasmota.ino:24:
tasmota/include/tasmota_compat.h:23:10: fatal error: esp8266toEsp32.h: No such file or directory

Here's the complete log from terminal

` Executing task: platformio run --target upload --target monitor --environment m5stack-atoms3 

Processing m5stack-atoms3 (platform: espressif32; board: m5stack-atoms3; framework: arduino)
--------------------------------------------------------------------------------------------------------------------------------------------------------
Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_QIO
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/m5stack-atoms3.html
PLATFORM: Espressif 32 (2024.5.13) > M5Stack AtomS3
HARDWARE: ESP32S3 240MHz, 320KB RAM, 8MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-builtin, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES: 
 - framework-arduinoespressif32 @ 3.0.0+sha.14a0ebd 
 - tool-esptoolpy @ 4.7.3 
 - tool-mkfatfs @ 2.0.1 
 - tool-mklittlefs @ 3.2.0 
 - tool-riscv32-esp-elf-gdb @ 12.1.0+20221002 
 - tool-xtensa-esp-elf-gdb @ 12.1.0+20221002 
 - toolchain-riscv32-esp @ 12.2.0+20230208 
 - toolchain-xtensa-esp32s3 @ 12.2.0+20230208
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 108 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0
|-- Adafruit MAX31865 library @ 1.0.1
|-- Adafruit MCP9808 Library @ 1.1.2
|-- Adafruit PM25 AQI Sensor @ 1.0.6
|-- Adafruit SGP30 Sensor @ 1.2.0
|-- Adafruit SGP40 Sensor @ 1.1.0
|-- Adafruit VEML7700 Library @ 1.0.0
|-- Adafruit seesaw Library @ 1.3.1
|-- ESP8266Audio @ 1.9.5
|-- BME8563
|-- EEPROM 24C128_256_521
|-- FFat @ 2.0.0
|-- FS @ 2.0.0
|-- I2Cdevlib-Core
|-- LibTeleinfo @ 1.1.7
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- NTP Library
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- SPL06-007 @ 0.1.0
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- headers
|-- WebServer @ 2.0.0
|-- ams @ 1.0
|-- arduino-mcp2515-1.0.1
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit TSL2591
|-- DNSServer @ 2.0.0
|-- DS2480 Library @ 0.1.0
|-- ESP8266SAM @ 1.0.1
|-- ESPFtpServer @ 0.1.0
|-- Ethernet @ 2.0.0
|-- FT5206_Library @ 1.0.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- GT911
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS @ 2.0.0
|-- LoRa @ 0.8.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- RadioLib @ 6.4.2
|-- Ext-printf @ 1.0
|-- SD @ 2.0.0
|-- SD_MMC @ 2.0.0
|-- SPI @ 2.0.0
|-- Sensirion I2C SEN5X @ 0.2.0
|-- TM1638plus @ 1.7.0
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 2.0.0
|-- Joba_Tsl2561 @ 2.0.10
|-- WiFiHelper @ 1.0.0
|-- WiFi @ 2.0.0
|-- Wire @ 2.0.0
|-- XPT2046_Touchscreen
|-- base64 @ 1.1.1
|-- BME68x
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ESP KNX IP Library @ 0.5.2
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- mp3_shine_esp32 @ 1.0.0
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
Building in release mode
Compiling .pio/build/m5stack-atoms3/src/tasmota.ino.cpp.o
Building .pio/build/m5stack-atoms3/bootloader.bin
Generating partitions .pio/build/m5stack-atoms3/partitions.bin
esptool.py v4.7.3
Creating esp32s3 image...
Merged 1 ELF section
Successfully created esp32s3 image.
Compiling .pio/build/m5stack-atoms3/liba7f/Wire/Wire.cpp.o
Compiling .pio/build/m5stack-atoms3/lib400/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp.o
Compiling .pio/build/m5stack-atoms3/lib6d7/SPI/SPI.cpp.o
In file included from /Users/edo/Downloads/Tasmota-development/tasmota/tasmota.ino:24:
tasmota/include/tasmota_compat.h:23:10: fatal error: esp8266toEsp32.h: No such file or directory

************************************************************************
* Looking for esp8266toEsp32.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:esp8266toEsp32.h"
* Web  > https://registry.platformio.org/search?q=header:esp8266toEsp32.h
*
************************************************************************

   23 | #include <esp8266toEsp32.h>
      |          ^~~~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio/build/m5stack-atoms3/src/tasmota.ino.cpp.o] Error 1
In file included from lib/lib_i2c/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp:1:
lib/lib_i2c/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.h:74:17: error: 'sint8_t' does not name a type; did you mean 'uint8_t'?
   74 |                 sint8_t begin(uint8_t addr = CCS811_ADDRESS);
      |                 ^~~~~~~
      |                 uint8_t
lib/lib_i2c/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp:10:1: error: 'sint8_t' does not name a type; did you mean 'uint8_t'?
   10 | sint8_t Adafruit_CCS811::begin(uint8_t addr)
      | ^~~~~~~
      | uint8_t
*** [.pio/build/m5stack-atoms3/lib400/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp.o] Error 1
============================================================= [FAILED] Took 23.12 seconds =============================================================

Environment     Status    Duration
--------------  --------  ------------
m5stack-atoms3  FAILED    00:00:23.115`

Can someone help me?

**Comment by Jason2866** [comment, marked as answer]
Tasmota has its own boards.json Do **not** use or add any others. This will not work.
The m5stack-atoms3 uses a S3 with QIO flash and no PSRAM. Just use the `[env:tasmota32s3]` to compile and it will work

**Comment by cturqueti** [comment]
Hello everyone,
I'm experiencing an issue while trying to compile Tasmota for the ESP32-S3 [env:tasmota32s3]. I keep getting the following error message:

C:/Users/Carlos/.platformio/packages/toolchain-xtensa-esp-elf/bin/../lib/gcc/xtensa-esp-elf/13.2.0/../../../../xtensa-esp-elf/bin/ld.exe: error: could not run lto-wrapper
collect2.exe: error: ld returned 1 exit status
*** [.pio\build\tasmota32s3-bluetooth\firmware.elf] Error 1

**Comment by Jason2866** [comment]
@cturqueti You run in the Windows path length limitation. Search Discussions how to work around the Windows only problem
